hicommonwealth/commonwealth
# Refactor version history (PART 1/2)

This will be the first ticket of two in an expand/contract database migration pattern. This is the expand portion where we will create the new entries for version_history, but not delete the old ones until part 2.

## Link to Issue
Closes: #7812

## Description of Changes
- Adds schema changes for new version_history to be added into their own tables
- Adds application level changes to add the new version_histories to this new table (While still keeping the original version_history functionality)
- Adds migration script that we will run async to reach parity with the new and old verion_history schemas.

## Test Plan
- Create thread (make sure `select * from "ThreadVersionHistories";` has correct data)
- Update thread (make sure `select * from "ThreadVersionHistories";` has correct data)
- Create comment (make sure `select * from "CommentVersionHistories";` has correct data)
- Update comment (make sure `select * from "CommentVersionHistories";` has correct data)

## Deployment Plan

1. Make a regular deployment
2. Wait for database migration to finish
3. Run ssh into heroku with `heroku run bash -a [appname]`  then in the packages/commonwealth directory run `pnpm run async-migrate-thread-version-history` and then  `pnpm run async-migrate-comment-version-history`
